### PR TITLE
Scoping azure list discovery to resource group

### PR DESCRIPTION
**Pull request by oxisto**
This PR adds support to the Azure listener to limit the scope to a certain resource group.

### CODE REVIEW

File: service/discovery/azure/azure.go [hunk 1]
```diff
@@ -205,3 +215,57 @@ func initClient[T any](existingClient *T, d *azureDiscovery, fun ClientCreateFun
 
 	return
 }
+
+// listPager is magic
```
[the remark this answers is not in the document]
Comment (oxisto) [reply service/discovery/azure/azure.go]: Can you have a look whether the documentation suffices?

File: service/discovery/azure/azure.go [hunk 2]
```diff
@@ -205,3 +215,77 @@ func initClient[T any](existingClient *T, d *azureDiscovery, fun ClientCreateFun
 
 	return
 }
+
+// listPager loops all values from a [runtime.Pager] object from the Azure SDK and issues a callback for each item. It
+// takes the following arguments:
+//   - d, an [azureDiscovery] struct,
+//   - newListAllPager, a function that supplies a [runtime.Pager] listing all resources of a specific Azure client,
+//   - newListByResourceGroupPager, a function that supplies a [runtime.Pager] listing all resource of a specific resource group,
```
Comment (anatheka) [inline service/discovery/azure/azure.go:223]:
```suggestion
//   - newListByResourceGroupPager, a function that supplies a [runtime.Pager] listing all resources of a specific resource group,
```

File: service/discovery/azure/azure.go [hunk 3]
```diff
@@ -205,3 +215,77 @@ func initClient[T any](existingClient *T, d *azureDiscovery, fun ClientCreateFun
...
+//   - resToValues1, a function that takes the response from a single page of newListAllPager and returns its values,
```
Comment (anatheka) [inline service/discovery/azure/azure.go:224]: Why do you vall it resToValues1 and resToValues2 and not resAll and resByResourceGroup are similiar?

**Review by anatheka — COMMENTED** [review]
Much better :)

**Review by anatheka — APPROVED** [review]
Perfect!

**Pull request merged**